org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -3983,13 +3997,31 @@ private void completionOnSingleNameReference(ASTNode astNode, ASTNode astNodePar
 
 			checkCancel();
 
-			findVariablesAndMethods(
-				this.completionToken,
-				scope,
-				singleNameReference,
-				scope,
-				insideTypeAnnotation,
-				singleNameReference.isInsideAnnotationAttribute);
+			// see if we can find argument type at position incase if we are at a vararg.
+			if (astNodeParent instanceof MessageSend m && this.expectedTypesPtr == -1) {
```

Comment:
Any chance you can pull some of the code here in new methods? `completionOnSingleNameReference` is already pretty long, this makes it even longer.